zwave_js_server/model/endpoint.py

```diff
@@ -64,3 +67,75 @@ def installer_icon(self) -> Optional[int]:
     def user_icon(self) -> Optional[int]:
         """Return user icon property."""
         return self.data.get("userIcon")
+
+    async def _async_send_command(
+        self,
+        cmd: str,
+        require_schema: Optional[int] = None,
+        wait_for_result: Optional[bool] = None,
+        **cmd_kwargs: Any,
+    ) -> Optional[Dict[str, Any]]:
+        """
+        Send an endpoint command. For internal use only.
+
+        If wait_for_result is not None, it will take precedence, otherwise we will decide to wait
+        or not based on the node status.
+        """
+        if self.client.driver is None:
+            raise FailedCommand(
```

Comment:
The library user may cause this error to be raised so we should test it.

Reply:
I'm trying to think how the user could hit this. The nodes and endpoints won't be available until the driver is created, and I don't see a place where we set the driver back to None when disconnecting. Should we set the driver to None when disconnecting since we would recreate it when we reconnect anyway (assuming I understand the logic correctly)?

Reply:
Yes, I think we should reset the driver on disconnect and also the result futures and shutdown complete event client attributes.

Reply:
oh boy. This is not as simple of a change as I thought. This might take me a bit to figure out

Reply:
OK I'm not quite sure how to test the new behavior of resetting everything explicitly because it doesn't appear that the mocks are set up to hit that part of the code at the moment. I've handled the scenario in the test by clearing the driver in the test itself, but I may need some help getting the additional coverage for the client